src/Client.php

```diff
@@ -160,16 +182,19 @@ protected function padCurlOptions(RequestInterface $request) : array
     {
         $options = $this->curlOptions;
 
-        $options[CURLOPT_RETURNTRANSFER] = true;
-        $options[CURLOPT_HEADER]         = true;
-        $options[CURLOPT_CUSTOMREQUEST]  = $request->getMethod();
-        $options[CURLOPT_URL]            = (string) $request->getUri();
-        $options[CURLOPT_POSTFIELDS]     = (string) $request->getBody();
+        $options[CURLOPT_RETURNTRANSFER] = $options[CURLOPT_RETURNTRANSFER] ?? true;
+        $options[CURLOPT_HEADER]         = $options[CURLOPT_HEADER] ?? true;
+        $options[CURLOPT_CUSTOMREQUEST]  = $options[CURLOPT_CUSTOMREQUEST] ?? $request->getMethod();
+        $options[CURLOPT_URL]            = $options[CURLOPT_URL] ?? (string) $request->getUri();
+        $options[CURLOPT_POSTFIELDS]     = $options[CURLOPT_POSTFIELDS] ?? (string) $request->getBody();
+        $options[CURLOPT_HTTPHEADER]     = $options[CURLOPT_HTTPHEADER] ?? [];
 
```

Comment:
this is semantically override if null, not override if falsey and not the same as override if invalid.

See https://repl.it/repls/TechnologicalDiligentWeblogsoftware for examples of each in an output table with headings.

Essentially you need a validator for every value type of curl option and to use that, not null coalesce

Reply:
here it is on an alternative (PHP 7.2) platform http://sandbox.onlinephpfunctions.com/code/45c40285a5d2127c3d2d0a7b24b3991ad3a9173a

Reply:
Original approach that the variable assigns a value directly without any validation or check.

You're right. It should check whether the variables is valid before assigning the value.